nifi-nar-bundles/nifi-hbase-bundle/nifi-hbase-processors/src/main/java/org/apache/nifi/hbase/DeleteHBaseRow.java

```diff
@@ -0,0 +1,154 @@
+            .expressionLanguageSupported(true)
+            .build();
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = super.getSupportedPropertyDescriptors();
+        properties.add(ROW_ID_LOCATION);
+        properties.add(FLOWFILE_FETCH_COUNT);
+        properties.add(BATCH_SIZE);
+        properties.add(KEY_SEPARATOR);
+
+        return properties;
+    }
+
+    @Override
+    protected void doDelete(ProcessContext context, ProcessSession session) throws Exception {
+        final int batchSize      = context.getProperty(BATCH_SIZE).asInteger();
+        final String location    = context.getProperty(ROW_ID_LOCATION).getValue();
+        final int flowFileCount  = context.getProperty(FLOWFILE_FETCH_COUNT).asInteger();
+        List<FlowFile> flowFiles = session.get(flowFileCount);
+
+        if (flowFiles != null && flowFiles.size() > 0) {
+            for (int index = 0; index < flowFiles.size(); index++) {
+                FlowFile flowFile = flowFiles.get(index);
+                try {
+                    if (location.equals(ROW_ID_BODY.getValue())) {
+                        doDeleteFromBody(flowFile, context, session, batchSize);
+                    } else {
+                        doDeleteFromAttribute(flowFile, context);
+                    }
+                    session.transfer(flowFile, REL_SUCCESS);
+                } catch (Exception ex) {
+                    getLogger().error(ex.getMessage(), ex);
+                    session.transfer(flowFile, REL_FAILURE);
+                }
+            }
+        }
+    }
+
+    private void doDeleteFromAttribute(FlowFile flowFile, ProcessContext context) throws Exception {
+        String rowKey = context.getProperty(ROW_ID).evaluateAttributeExpressions(flowFile).getValue();
+        String tableName = context.getProperty(TABLE_NAME).evaluateAttributeExpressions(flowFile).getValue();
+        clientService.delete(tableName, rowKey.getBytes("UTF-8"));
+    }
+
+    private void doDeleteFromBody(FlowFile flowFile, ProcessContext context, ProcessSession session, int batchSize) throws Exception {
+        String keySeparator = context.getProperty(KEY_SEPARATOR).evaluateAttributeExpressions(flowFile).getValue();
+        String tableName = context.getProperty(TABLE_NAME).evaluateAttributeExpressions(flowFile).getValue();
+
+        List<byte[]> deletes = new ArrayList<>();
+        session.read(flowFile, in -> {
+            String data = IOUtils.toString(in, "UTF-8");
+            if (data != null && data.length() > 0) {
+                String[] parts = data.split(keySeparator);
+                for (int index = 0; index < parts.length; index++) {
+                    deletes.add(parts[index].getBytes("UTF-8"));
+                }
+            }
+        });
+
+
+        List<byte[]> batch = new ArrayList<>();
+        for (int index = 0; index < deletes.size(); index++) {
```

Comment:
this for loop can be avoided and it can be merged in the previous one, can't it?

Reply:
I'll take a closer look.